src/common/transformations/src/transformations/op_conversions/eye_decomposition.cpp

```diff
@@ -0,0 +1,123 @@
+    const auto eye_size = register_new_node<opset9::ReduceMin>(shape_pad_diff, zero_int, true);
+    const auto pad_end = register_new_node<opset9::Subtract>(shape_pad_diff, eye_size);
+
+    // Make 1d-eye as eye_size times of (1, zeros(eye_size)), trimmed at end by eye_size elements.
+    const auto zeros = register_new_node<opset9::Tile>(zero, eye_size);
+    const auto one_followed_by_zeros = register_new_node<opset9::Concat>(OutputVector{one, zeros}, 0);
+    const auto eye_1d = register_new_node<opset9::Pad>(register_new_node<opset9::Tile>(one_followed_by_zeros, eye_size),
+                                                       zero_int,
+                                                       register_new_node<opset9::Negative>(eye_size),
+                                                       op::PadMode::CONSTANT);
+    // Reshape 1d-eye to 2d-eye
+    const auto eye_2d =
+        register_new_node<opset9::Reshape>(eye_1d,
+                                           register_new_node<opset9::Concat>(OutputVector{eye_size, eye_size}, 0),
+                                           false);
+
+    // Pad Eye to get final shape
+    return register_new_node<opset9::Pad>(eye_2d, pad_start, pad_end, op::PadMode::CONSTANT);
+}
+
+std::shared_ptr<Node> EyeDecomposition::make_eye_batches(const Output<Node>& eye, const Output<Node>& batch) {
+    const auto eye_tile = register_new_node<opset9::Constant>(element::i64, Shape{2}, 1);
+
+    // `batch_repeats` repeat eye matrix as tile only in higher dimensions than 1 by number(s) in batch parameter.
+    const auto batch_repeats = register_new_node<opset9::Concat>(OutputVector{batch, eye_tile}, 0);
+
+    return register_new_node<opset9::Tile>(eye, batch_repeats);
+}
+
+EyeDecomposition::EyeDecomposition() {
+    auto p_height = pattern::any_input();
+    auto p_width = pattern::any_input();
+    auto p_k = pattern::wrap_type<opset9::Constant>(k_predicate);
+    auto p_batch = pattern::wrap_type<opset9::Constant>(batch_predicate);
+
+    auto p_eye_no_batch = pattern::wrap_type<opset9::Eye>({p_height, p_width, p_k});
+    auto p_eye_batch = pattern::wrap_type<opset9::Eye>({p_height, p_width, p_k, p_batch});
+
+    auto p_eye = std::make_shared<pattern::op::Or>(OutputVector{p_eye_batch, p_eye_no_batch});
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto m_eye = std::dynamic_pointer_cast<opset9::Eye>(m.get_match_root());
+
+        if ((!m_eye) || transformation_callback(m_eye)) {
+            return false;
+        }
+
+        const auto& pattern_to_output = m.get_pattern_value_map();
+
+        const auto dtype = m_eye->get_out_type();
+        const auto width = pattern_to_output.at(p_width);
+        const auto height = pattern_to_output.at(p_height);
+        const auto k = pattern_to_output.at(p_k);
+
+        auto eye = make_eye_model(height, width, k, dtype);
+
+        if (m_eye->get_input_size() == p_eye_batch->get_input_size()) {
```

Comment:
maybe it's better to check that p_batch is actually exists in the pattern map?
auto batch = pattern_to_output.find(p_batch);
if (batch != pattern_to_output.end()) {
...